src/shared/sleep-config.c

```diff
@@ -32,31 +32,44 @@
 #include "strv.h"
 #include "time-util.h"
 
-int parse_sleep_config(const char *verb, bool *ret_allow, char ***ret_modes, char ***ret_states, usec_t *ret_delay) {
-        int allow_suspend = -1, allow_hibernate = -1,
-            allow_s2h = -1, allow_hybrid_sleep = -1;
-        bool allow;
-        _cleanup_strv_free_ char
-                **suspend_mode = NULL, **suspend_state = NULL,
-                **hibernate_mode = NULL, **hibernate_state = NULL,
-                **hybrid_mode = NULL, **hybrid_state = NULL;
-        _cleanup_strv_free_ char **modes, **states; /* always initialized below */
-        usec_t delay = 180 * USEC_PER_MINUTE;
+SleepConfig *sleep_config_new(void) {
+        SleepConfig *sc = NULL;
+
+        sc = new(SleepConfig, 1);
+        if (!sc)
+                return NULL;
+
+        *sc = (SleepConfig) {
+                .allow_suspend       = -1,
+                .allow_hibernate     = -1,
+                .allow_s2h           = -1,
+                .allow_hybrid_sleep  = -1,
+        };
+
+        return sc;
+}
+
+int parse_sleep_config(const char *verb, SleepConfig **ret_sleep_config) {
```

Comment:
Before, the verb was used to decide which set of modes, states, and allow value was returned. Now you're always parsing everything, and the verb is unused. Drop it.

Reply:
Glad this is the case. I'd left it here because I wasn't entirely sure if the original logic that checked the verb and modes/states before returning `log_oom` was necessary. Happy to remove the verb.